DataFormats/ForwardDetId/interface/HGCalDetId.h

```diff
@@ -59,6 +59,27 @@ class HGCalDetId : public DetId {
   static const HGCalDetId Undefined;
 };
 
+namespace HGCalDetIdUtil
+{
+  // Functions to work on raw HGCAlDetIds
+  int subdetIdOf(uint32_t id);
+  int cellOf(uint32_t id);
+  int waferOf(uint32_t id);
+  int waferTypeOf(uint32_t id);
+  int layerOf(uint32_t id);
+  int zsideOf(uint32_t id);
+
+  void setCellOf(uint32_t& id, int cell);
+  void setWaferOf(uint32_t& id, int mod);
```

Comment:
@jbsauvan @davidlange6 Actually if you are passing uint32_t you should pass by value (32 bits wide), rather than by reference (64 bits wide on most systems nowadays).